inst/include/Rcpp/api/meat/DottedPairImpl.h

```diff
@@ -53,7 +53,7 @@ namespace Rcpp{
 		} else {
 			if( ref.isNULL( ) ) throw index_out_of_bounds() ;
 
-			if( static_cast<R_len_t>(index) > ::Rf_length(ref.get__()) ) throw index_out_of_bounds() ;
+                        if( static_cast<R_xlen_t>(index) > ::Rf_length(ref.get__()) ) throw index_out_of_bounds() ;
```

Comment:
Here you cast to `R_xlen_t` but check against `::Rf_length`. Is that intentional? Should we be using `::Rf_xlength`?